common/views.py

```diff
@@ -0,0 +1,73 @@
+from django.contrib.auth.models import AnonymousUser
+
+from rest_framework import generics
+
+from .models import (
+    Contact, County, SubCounty, Constituency)
+
+from .serializers import (
+    ContactSerializer, CountySerializer, SubCountySerializer,
+    ConstituencySerializer)
+
+
+class FilterViewMixin(object):
```

Comment:
Good start, but this is still a blunt instrument. 

It will be more maintainable if done via a custom permissions class [ http://www.django-rest-framework.org/api-guide/permissions/#custom-permissions ] that can be added to the views that need it ( while other views are left with the default ). The internal implementation of that custom permissions class will be very similar to the code you have here.

The code you have below is also "too branchy". It could be rewritten in a shorter way as follows:

```python
def get_queryset(self):
    user = self.request.user
    if isinstance(user, AnonymousUser) or user.is_national:
        return self.queryset  # Leave the queryset unbothered
    elif user.is_incharge:
       return self.queryset.filter(county=user.county)  # Blind spot; what about sub-counties?
    else:
        return [] # Isn't it better to raise a custom exception here, rather than fail silently. This should not happen with a properly set up account
```

The rewritten sketch above makes it easy to follow the logic for each of the three possible "exit points".